proxy/coco/src/peer/gossip.rs

```diff
@@ -20,13 +21,14 @@ pub async fn announce(state: &State, urn: &RadUrn, rev: Option<Oid>) {
 }
 
 /// Emit a [`Gossip`] request for the given `urn`.
-pub async fn query(state: &State, urn: RadUrn) {
-    let protocol = state.api.protocol().clone();
-    protocol
+pub async fn query(state: &State, urn: RadUrn, origin: Option<PeerId>) {
```

Comment:
```suggestion
pub async fn query(state: &State, urn: RadUrn, origin: impl Into<Option<PeerId>>) {
```

This allows us to pass a `PeerId` without wrapping it in `Some`. Not a blocker, but I do like the pattern

Reply:
I’m going to leave it as it is since the other code in this file does not use the pattern.